internal/manifests/collector/config_replace.go

```diff
@@ -33,12 +34,57 @@ type Config struct {
 }
 
 func ReplaceConfig(instance v1alpha1.AmazonCloudWatchAgent) (string, error) {
+	// Parse the original configuration from instance.Spec.Config
 	config, err := adapters.ConfigFromJSONString(instance.Spec.Config)
 	if err != nil {
 		return "", err
 	}
 
-	out, err := json.Marshal(config)
+	conf := confmap.NewFromStringMap(config)
+
```

Comment:
You would also need this check because if it is empty, then the `/etc/prometheusconfig/prometheus.yaml` won't be created.
```go
!instance.Spec.Prometheus.IsEmpty()
```

Reply:
done